adapter/internal/utils/string_utils.go

```diff
@@ -0,0 +1,40 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package stringutils
+
+import (
+	"strings"
+)
+
+// MaskString takes string as argument and keep a visibleLength number of charators and mask rest of the string with maskCharacter
+// It can mask the charators from either left or right hand side of the string depending on maskRight boolean
+// It returns the masked string
+func MaskString(str string, visibleLength int, maskCharacter string, maskRight bool) string {
```

Comment:
Shall we create a new function `MaskToken` with some default values and it calls this function with that default values. i.e. MaskToken calls -> MaskString(token, 4, "*", false), So we can have kind of masking in other places.